sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageBody.java

```diff
@@ -3,14 +3,113 @@
+ * corresponding to the type of the body throws exception.</b>
+ *
+ * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+ * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+ *
+ * @see AmqpMessageBodyType
  */
-public interface AmqpMessageBody {
+public final class AmqpMessageBody {
+    private final ClientLogger logger = new ClientLogger(AmqpMessageBody.class);
+    private AmqpMessageBodyType bodyType;
+
+    private List<byte[]> data;
+
+    AmqpMessageBody() {
+        // package constructor so no one can create instance of this except classes im this package.
+    }
+
+    /**
+     * Creates instance of {@link AmqpMessageBody} with given byte array.
+     *
+     * @param data used to create another instance of {@link AmqpMessageBody}.
+     *
+     * @return AmqpMessageBody Newly created instance.
+     *
+     * @throws NullPointerException if {@code data} is null.
+     */
+    public static AmqpMessageBody fromData(byte[] data) {
+        Objects.requireNonNull(data, "'data' cannot be null.");
+        AmqpMessageBody body = new AmqpMessageBody();
+        body.bodyType = AmqpMessageBodyType.DATA;
+        body.data = Collections.singletonList(Arrays.copyOf(data, data.length));
+        return body;
+    }
+
     /**
-     * Type representing various supported amqp body types.
+     * Gets the {@link AmqpMessageBodyType} of the message.
+     * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+     * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+     * @return AmqpBodyType type of the message.
+     */
+    public AmqpMessageBodyType getBodyType() {
+        return bodyType;
+    }
+
+    /**
+     * Gets an {@link IterableStream} of byte array containing only first byte array set on this
+     * {@link AmqpMessageBody}. This library only support one byte array, so the returned list will have only one
+     * element. Look for future releases where we will support multiple byte array.
      *
-     * @return The {@link AmqpBodyType}.
+     * <b>Client should test for {@link AmqpMessageBodyType} before calling corresponding get method.Get methods not
+     * corresponding to the type of the body throws exception.</b>
+     *
+     * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+     * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+     * @return data set on {@link AmqpMessageBody}.
+     *
+     * @throws IllegalArgumentException If {@link AmqpMessageBodyType} is not {@link AmqpMessageBodyType#DATA DATA}.
      */
-    AmqpBodyType getBodyType();
+    public IterableStream<byte[]> getData() {
+        if (bodyType != AmqpMessageBodyType.DATA) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(String.format("Cannot return data for a "
+                + "message which is of type %s.", getBodyType().toString())));
```

Comment:
This kind of error message only serves to confuse users. It is better to say that the type is not supported at present and that it will be available in future releases. Perhaps even going so far as to point to a pre-filed github issue so that users may find more or subscribe.